postgres-openssl/src/lib.rs

```diff
@@ -134,13 +134,15 @@ where
     S: AsyncRead + AsyncWrite + Unpin + Debug + 'static + Sync + Send,
 {
     type Stream = TlsStream<S>;
-    type Error = HandshakeError<S>;
+    type Error = Error;
     #[allow(clippy::type_complexity)]
-    type Future = Pin<Box<dyn Future<Output = Result<TlsStream<S>, HandshakeError<S>>> + Send>>;
+    type Future = Pin<Box<dyn Future<Output = Result<TlsStream<S>, Error>> + Send>>;
 
     fn connect(self, stream: S) -> Self::Future {
         let future = async move {
-            let stream = tokio_openssl::connect(self.ssl, &self.domain, stream).await?;
+            let ssl = self.ssl.into_ssl(&self.domain)?;
+            let mut stream = Box::pin(SslStream::new(ssl, stream)?);
+            stream.as_mut().connect().await?;
```

Comment:
I think it's worth doing something just a little bit more complicated to get better diagnostics after a certificate validation failure: https://github.com/sfackler/hyper-openssl/blob/master/src/lib.rs#L237-L240

Reply:
SGTM! I just shamelessly copy/pasted that code into here.